library/analytics/src/main/java/org/cru/godtools/analytics/firebase/FirebaseAnalyticsService.kt

```diff
@@ -54,43 +55,34 @@ class FirebaseAnalyticsService @VisibleForTesting internal constructor(
     @MainThread
     @Subscribe(threadMode = ThreadMode.MAIN_ORDERED)
     fun onAnalyticsEvent(event: AnalyticsBaseEvent) {
-        if (event.isForSystem(AnalyticsSystem.FIREBASE)) when (event) {
+        if (event.isForSystem(AnalyticsSystem.FIREBASE) || event.isForSystem(AnalyticsSystem.ADOBE)) when (event) {
             is AnalyticsScreenEvent -> handleScreenEvent(event)
             is AnalyticsActionEvent -> handleActionEvent(event)
         }
     }
 
-    // region ActivityLifecycleCallbacks
-    init {
-        app.registerActivityLifecycleCallbacks(this)
-    }
-
-    private var currentActivity: Activity? by weak()
-
-    override fun onActivityResumed(activity: Activity) {
-        currentActivity = activity
-    }
-
-    override fun onActivityPaused(activity: Activity) {
-        currentActivity = currentActivity?.takeUnless { it === activity }
-    }
-
-    override fun onActivityCreated(activity: Activity, savedInstanceState: Bundle?) = Unit
-    override fun onActivityStarted(activity: Activity) = Unit
-    override fun onActivityStopped(activity: Activity) = Unit
-    override fun onActivitySaveInstanceState(activity: Activity, outState: Bundle) = Unit
-    override fun onActivityDestroyed(activity: Activity) = Unit
-    // endregion ActivityLifecycleCallbacks
-    // endregion Tracking Events
-
     @MainThread
     private fun handleScreenEvent(event: AnalyticsScreenEvent) {
-        currentActivity?.let { firebase.setCurrentScreen(it, event.screen, null) }
+        val bundle = Bundle().apply {
+            putString(FirebaseAnalytics.Param.SCREEN_NAME, event.screen)
+            putString("cru_appname", VALUE_GODTOOLS)
+            event.locale?.let { putString("cru_contentlanguage", LocaleCompat.toLanguageTag(it)) }
+            event.adobeSiteSection?.let { putString("cru_sitesection", event.adobeSiteSection) }
+            event.adobeSiteSubSection?.let { putString("cru_sitesubsection", event.adobeSiteSubSection) }
+        }
+        firebase.logEvent(FirebaseAnalytics.Event.SCREEN_VIEW, bundle)
     }
 
     @MainThread
     private fun handleActionEvent(event: AnalyticsActionEvent) {
-        firebase.logEvent(event.firebaseEventName, null)
+        val bundle = Bundle().apply {
+            event.adobeAttributes?.forEach { attribute ->
```

Comment:
I'd like us to move away from adobe specific properties on the event objects. So we should probably create a `firebaseAttributes` property on the event object that exposes the attributes in the correct format.

Reply:
hmm, `adobeSiteSection` and `adobeSiteSubSection` would also fall under the renaming need. This can be done in a followup PR if you would like

Reply:
So should I leave the adobeAttributes as is, or put in the effort now to change it? It could also be done with all the adobe clean work that will eventually come.

Reply:
let's do it with all the adobe cleanup work later

Reply:
when we do that work I will want to get rid of the replace calls at that time